src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -355,6 +356,19 @@ public void edit() {
         });
     }
 
+    public void renameFile() {
+        LinkedFileRenameDialogView dialog = new LinkedFileRenameDialogView(this.linkedFile);
```

Comment:
I'm really sorry for your work, but in my opinion a simple TextInputDialog should be sufficient (instead of a custom dialog): https://github.com/JabRef/jabref/blob/9ab59d0444916b51ff6969be4adaa342291f2964/src/main/java/org/jabref/gui/JabRefDialogService.java#L132

Reply:
@tobiasdiez  A TextInputDialog does not have a browse button....